src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+                individualOverlapInfo.meanOverlapMappingQuality, individualOverlapInfo.numCoherent,
+                individualOverlapInfo.coherentMappingQuality,
+                clusterOverlapInfo.numOverlap, clusterOverlapInfo.overlapMappingQuality,
+                clusterOverlapInfo.meanOverlapMappingQuality, clusterOverlapInfo.numCoherent,
+                clusterOverlapInfo.coherentMappingQuality,
+                referenceGapOverlap, umapS100
+            }
+        );
+    }
+
+    private int getMappingQuality(final BreakpointEvidence evidence) {
+        // Note: return "max" mapping quality for non-ReadEvidence. Reasoning: some features depend on sum or average of
+        // read qualities. Non-ReadEvidence isn't *bad* per se, so give it a good score.
+        return evidence instanceof BreakpointEvidence.ReadEvidence ?
+            ((BreakpointEvidence.ReadEvidence) evidence).getMappingQuality() : 60;
+    }
+
+    private double getTemplateSize(final BreakpointEvidence evidence) {
+        if(evidence instanceof BreakpointEvidence.ReadEvidence) {
+            // For ReadEvidence, return templateSize as percentile of library's cumulative density function:
+            final String readGroup = ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup();
+            final String library = readGroupToLibraryMap.get(readGroup);
+            final LibraryStatistics libraryStatistics = libraryStatisticsMap.get(library);
+            final IntHistogram.CDF templateSizeCDF = libraryStatistics.getCDF();
+            /*
+            final IntHistogram.CDF templateSizeCDF = libraryStatisticsMap.get(
+                    ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup()
+            ).getCDF();
+            */
+            final int cdfBin = Integer.min(Math.abs(((BreakpointEvidence.ReadEvidence) evidence).getTemplateSize()),
+                                           templateSizeCDF.size() - 1);
+            return templateSizeCDF.getFraction(cdfBin);
+        } else if(evidence instanceof BreakpointEvidence.TemplateSizeAnomaly) {
+            // For TemplateSizeAnomaly, return readCount scaled by coverage
+            return (double)(((BreakpointEvidence.TemplateSizeAnomaly) evidence).getReadCount()) / coverage;
+        } else {
+            throw new IllegalArgumentException(
+                    "templateSize feature only defined for ReadEvidence and TemplateSizeAnomaly"
+            );
+        }
+    }
+
+    /**
+     * For given BreakpointEvidence, calculate number of overlappers, coherent pieces of BreakpointEvidence, and
+     * the sums of their mapping qualities. Cache this information so that it can be looked up when computing max or
+     * average of these values over the set of overlapping evidence (as in getClusterOverlapInfo).
+     */
+    private void cacheOverlapInfo(final BreakpointEvidence evidence) {
+        int numOverlap = 0;
```

Comment:
Personally, I would make these `final MutableInt`s and use `increment` below.

Reply:
I don't see the advantage. I'd be replacing a non-final primitive with a final class object wrapping a non-final primitive. I did change the python-like += 1 to a proper increment though.